src/events/nats.js

```diff
@@ -0,0 +1,170 @@
+/* eslint-disable camelcase */
+/**
+ * @fileoverview Instrumentation for nats library.
+ */
+const shimmer = require('shimmer');
+const tracer = require('../tracer.js');
+const eventInterface = require('../event.js');
+const moduleUtils = require('./module_utils.js');
+
+const NATS_TYPES = {
+    name: 'nats',
+    mainWrappedFunction: 'Client',
+    badMessage: 'NATS_BAD_JSON_MSG',
+};
+
+const getServerHostname = (currentServer) => {
+    let serverHostname = NATS_TYPES.serverDefaultHostname;
+    if (currentServer.url && currentServer.url.hostname) {
+        serverHostname = currentServer.url.hostname;
+    }
+    return serverHostname;
+};
+
+const getPublishParams = (subject, msg, opt_reply, opt_callback, jsonConnectProperty) => {
```

Comment:
is this logic copied from nuts (the best answer would be yes)

Reply:
yes